packages/nodejs/README.md

```diff
@@ -0,0 +1,512 @@
+...
+```
+
+## getPoolStatus
+Retrieves the status of a pool (either a specific baker or passive delegation) at a specific block.
+If a baker ID is specified, the status of that baker is returned. To get the status of passive delegation, baker ID should be left undefined.
+```js
+const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";
+const bakerId = BigInt(1);
+
+const bakerStatus = await client.getPoolStatus(blockHash, bakerId);
+const passiveDelegationStatus = await client.getPoolStatus(blockHash); 
+...
+```
+
+## getRewardStatus
+Retrieves the current amount of funds in the system at a specific block, and the state of the special accounts. 
+```js
+const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";
+
+const rewardStatus = await client.getRewardStatus(blockHash);
+```
+
+Protocol version 4 expanded the amount of information in the response, so one should check the type to access that.
+This information includes information about the payday and total amount of funds staked.
+```js
+if (isRewardStatusV1(rewardStatus)) {
+    const nextPaydayTime = rewardStatus.nextPaydayTime; 
+    ...
+}
+```
+
+## Check block for transfers with memo
+The following example demonstrates how to check and parse a block 
+for transfers with a memo.
+```js
+const blockHash = "b49bb1c06c697b7d6539c987082c5a0dc6d86d91208874517ab17da752472edf";
+const blockSummary = await client.getBlockSummary(blockHash);
+const transactionSummaries = blockSummary.transactionSummaries;
+
+for (const transactionSummary of transactionSummaries) {
+    if (transactionSummary.result.outcome === 'success') {
+        if (instanceOfTransferWithMemoTransactionSummary(transactionSummary)) {
+            const [transferredEvent, memoEvent] = transactionSummary.result.events;
+
+            const toAddress = transferredEvent.to.address;
+            const amount = transferredEvent.amount;
+            const memo = memoEvent.memo;
+
+            // Apply business logic to toAddress, amount and memo...
+        }
+    }
+}
+```
+
+## getInstances
+Used to get the full list of contract instances on the chain at a specific block.
+```js
+const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";
+
+const instances = await client.getInstances(blockHash);
+...
+```
+
+## getInstanceInfo
+Used to get information about a specific contract instance, at a specific block.
+
+```js
+const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";
+const contractAddress = { index: 1n, subindex: 0n };
+
+const instanceInfo = await client.getInstanceInfo(blockHash, contractAddress);
+const name = instanceInfo.name; 
+...
+```
+
+Note that only version 0 contracts returns the model. (use `isInstanceInfoV0`/`isInstanceInfoV1` to check the version)
+
+# Build
+
+## Building for a release
+To build the package run
+```
+yarn build
+```
+
+Note that this also builds the [common package](../common).
```

Comment:
This does not seem to be the case. If I run `yarn build` for the `nodejs` package, then it fails if `common` has not been built.

Reply:
Yeah, I changed the text, to say you need to built it first.